Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectReductionAndAnalysis.py

```diff
@@ -213,8 +247,22 @@ def PyExec(self):
             msd_alg.setProperty("OutputWorkspace", self._scan_ws + '_msd')
             msd_alg.setProperty("FitWorkspaces", self._scan_ws + '_msd_fit')
             msd_alg.execute()
-            mtd.addOrReplace(self._scan_ws + '_msd', msd_alg.getProperty("OutputWorkspace").value)
-            mtd.addOrReplace(self._scan_ws + '_msd_fit', msd_alg.getProperty("FitWorkspaces").value)
+
+        if self._diffraction:
+            diff_prog = Progress(self, start=0.9,end=1.0,nreports=1)
+            diff_prog.report('Diffraction Reduction')
+            diff_red_alg = self.createChildAlgorithm("ISISIndirectDiffractionReduction", enableLogging=False)
+            diff_red_alg.setProperty('InputFiles', self._data_files)
+            diff_red_alg.setProperty('ContainerFiles',self._can_files)
+            diff_red_alg.setProperty('SumFiles', self._sum_files)
+            diff_red_alg.setProperty('LoadLogFiles', self._load_logs)
+            diff_red_alg.setProperty('Instrument', self._instrument_name)
+            diff_red_alg.setProperty('Mode', self._diff_mode)
+            diff_red_alg.setProperty('SpectraRange',self._diff_spec)
+            diff_red_alg.setProperty('GroupingPolicy', self._grouping_method)
+            diff_red_alg.setProperty('OutputWorkspace', self._diff_workspace)
+            diff_red_alg.execute()
+            self.setProperty("DiffractionWorkspace", diff_red_alg.getProperty("OutputWorkspace").value)
```

Comment:
In fact I'd go for one method per algorithm i.e. something like `self.run_diffraction_reduction()`, `self.run_msg()` ...